File: bindings/nodejs/examples/how_tos/accounts_and_addresses/create-mnemonic.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Utils } from '@iota/sdk';

// Run with command:
// yarn run-example ./how_tos/accounts_and_addresses/create-mnemonic.ts

// In this example we will generate a random BIP39 mnemonic
async function run() {
    try {
        const mnemonic = Utils.generateMnemonic();
        console.log('Mnemonic: ' + mnemonic);
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());

File: sdk/examples/how_tos/wallet/create_mnemonic.rs
// Copyright 2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

//! In this example we will create a random mnemonic.
//!
//! ```sh
//! `cargo run --release --all-features --example create_mnemonic`
//! ```

use iota_sdk::client::Client;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mnemonic = Client::generate_mnemonic()?;

    println!("Generated mnemonic:\n{}", mnemonic.as_ref());

    Ok(())
}
